### DISCUSSION TITLE
Port status LEDs

### DISCUSSION THREAD

**Discussion by dberlin80** [darthcloud/BlueRetro#703] [General]
Hello,
I made myself a blueretro adapter for each my PS2 and Dreamcast which work fine with my bluetooth controllers. Great project, thanks :)

Now I wanted to get a step further and add the port status LEDs.
The issue:
The LED on IO2 works fine, pulses and gets solid when paired. But the LEDs on the other IO ports don‘t pulse or light up. They remain OFF. I tested all other LEDs on IO2 so they’re good. 
Any hint?
Thanks much.

**Comment by icd2k3** [comment]
I noticed they'll stay off (with the exception of port 1) unless you manually put it in pairing mode - but if you connect a 2nd controller that IO _**should**_ light up solid.

**Reply by dberlin80** [reply]
Thx much, I'll look into that and will reply.

**Reply by dberlin80** [reply]
Ok, so I tested everything out and it was the just as you described thx :)

**Reply by GrumpyGopher** [reply]
@dberlin80 when you wired what MOSFET did you use?  Do you have a part number?
thanks

**Reply by dberlin80** [reply]
@tr12345 
Darthcloud mentioned somewhere that you don’t necessarily need one to have it working. By now I made two- one with and one without and he‘s right. If you want to make it properly however, then you can use any standard nmosfet- I used this:

BSS138LT1G
https://a.aliexpress.com/_Ezc24RD

**Reply by GrumpyGopher** [reply]
Yeah I saw that post but had trouble getting it to work.  How did you wire it up without?

**Reply by dberlin80** [reply]
I‘m honest… I forgot which wiring I exactly used as I closed both systems(built it inside the consoles) but it was either step 7 here:
https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-DIY-Build-Instructions

Or the actual way intended with a mosfet but where I just left it out.

**Comment by darthcloud** [comment]
if you skip the mosfet wire like this
![test](https://github.com/darthcloud/BlueRetro/assets/3744056/2f7f63df-3d43-441b-a548-b008cd3aaf20)
3.3v is the esp32 pins

0v is GND

**Reply by GrumpyGopher** [reply]
Ok I got this to work for all 4 now.  I'm just using a 100 ohm resistor and I swear my LED is the other way around but it's working.

I was also doing something stupid where I had the firmware set to universal but it wasn't plugged into a system (I'm still waiting on some parts) this was causing the P1 LED to just be always on.  I found if I went int the config and selected Dreamcast or If I programmed it with the Dreamcast firmware all lights worked as expected.

Thanks for the help everyone!

**Reply by GrumpyGopher** [reply]
I stand corrected the LED was oriented the way you said.  Here is the diagram of how I have it connected
![image](https://github.com/darthcloud/BlueRetro/assets/27497732/c5d94a25-4ef3-4939-acd8-0b08ba9b1042)
